nighthawk/test/client_test.cc

```diff
@@ -0,0 +1,148 @@
+public:
+  ClientTest() : Envoy::BaseIntegrationTest(GetParam(), realTime(), ClientTest::envoy_config) {}
+
+  static void SetUpTestCase() {
+    Envoy::Filesystem::InstanceImplPosix file_system;
+    envoy_config = file_system.fileReadToEnd(Envoy::TestEnvironment::runfilesPath(
+        "nighthawk/test/test_data/benchmark_http_client_test_envoy.yaml"));
+    envoy_config = Envoy::TestEnvironment::substitute(envoy_config);
+  }
+
+  void SetUp() override {
+    // We fork the integration test fixture into a child process, to avoid conflicting
+    // runtimeloaders as both NH and the integration server want to own that and we can have only
+    // one.
+    pipe(fd_port);
+    pipe(fd_confirm);
+    pid_ = fork();
+    ASSERT(pid_ >= 0);
+
+    if (pid_ == 0) {
+      // child process running the integration test server.
+      ares_library_init(ARES_LIB_INIT_ALL);
+      Envoy::Event::Libevent::Global::initialize();
+      initialize();
+      int port = lookupPort("listener_0");
+      int parent_message;
+      write(fd_port[1], &port, sizeof(port));
+      // The parent process writes to fd_confirm when it has read the port. This call to read blocks
+      // until that happens.
+      read(fd_confirm[0], &parent_message, sizeof(parent_message));
+      ASSERT(parent_message == 1234);
+      // The parent process closes fd_port when the test tears down. The read call blocks until it
+      // does that.
+      ASSERT(read(fd_port[0], &port_, sizeof(port_)) == -1);
+    } else if (pid_ > 0) {
+      ASSERT(read(fd_port[0], &port_, sizeof(port_)) > 0);
+      ASSERT(port_ > 0);
+      int foo = 1234;
+      write(fd_confirm[1], &foo, sizeof(foo));
+    }
+  }
+
+  void TearDown() override {
+    if (pid_ == 0) {
+      test_server_.reset();
+      fake_upstreams_.clear();
+      ares_library_cleanup();
+    }
+    close(fd_confirm[0]);
+    close(fd_confirm[1]);
+    close(fd_port[0]);
+    close(fd_port[1]);
+  }
+
+  std::string testUrl() {
+    ASSERT(pid_ > 0);
+    const std::string address = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+    return fmt::format("http://{}:{}/", address, port_);
+  }
+
+  std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+    std::vector<std::string> words = Envoy::TestUtility::split(args, ' ');
+    std::vector<const char*> argv;
+    for (const std::string& s : words) {
+      argv.push_back(s.c_str());
+    }
+    return std::make_unique<OptionsImpl>(argv.size(), argv.data());
+  }
+
+  int port_;
+  pid_t pid_;
+  int fd_port[2];
+  int fd_confirm[2];
+  static std::string envoy_config;
+};
+
+std::string ClientTest::envoy_config;
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ClientTest,
+                         testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
+TEST_P(ClientTest, NormalRun) {
+  if (pid_ > 0) {
```

Comment:
Can't we kill off the `fork`ed gtest stuff at setup time, rather than in each test here?

Reply:
Yes, turns out we could. Fixed by using `SKIP_TEST` in the test setup.